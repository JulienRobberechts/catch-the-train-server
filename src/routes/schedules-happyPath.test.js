const request = require("supertest");
const debug = require("debug")("ctt:test");

const app = require("../app");
const api = request(app);
const { mockApiCalls } = require("../tests/mock/ratp-api/mockApiCalls");

const config = require("../config");

describe("GET /next-train/{network}/{line}/{station}?missions=...", () => {
  const station = "chatelet+les+halles";
  const url = "/next-trains/rers/A/" + station;
  beforeEach(() => {
    if (config.RATP_API_MOCK_DATA)
      throw Error("RATP_API_MOCK_DATA should be set to 'false' for this test");
    mockApiCalls(station);
  });
  it("should return all missions for chatelet", async () => {
    const response = await api.get(url).expect(200);
    expect(response.body.departures).toMatchSnapshot();
    expect(response.body.departures.length).toBe(8);
  });
  it("should return 'UPAC' missions for chatelet", async () => {
    const response = await api.get(url + "?missions=UPAC").expect(200);
    expect(response.body.departures).toMatchSnapshot();
    expect(response.body.departures.length).toBe(2);
  });
  it("should return 'UPAC' and 'ZEBU' missions for chatelet", async () => {
    const response = await api.get(url + "?missions=UPAC,ZEBU").expect(200);
    expect(response.body.departures).toMatchSnapshot();
    expect(response.body.departures.length).toBe(5);
  });
});

describe("GET schedule by destination", () => {
  const fromStation = "chatelet+les+halles";
  beforeEach(() => {
    if (config.RATP_API_MOCK_DATA)
      throw Error("RATP_API_MOCK_DATA should be set to 'false' for this test");
    mockApiCalls(fromStation);
  });
  it("should return 5 departures from chatelet+les+halles to maisons+laffitte", async () => {
    const toStation = "maisons+laffitte";
    const url = `/next-trains/rers/A/${fromStation}/${toStation}`;
    const response = await api.get(url).expect(200);
    expect(response.body.departures).toMatchSnapshot();
    expect(response.body.departures.length).toBe(5);
  });
  it("should return 2 departures from chatelet+les+halles to cergy+le+haut", async () => {
    const toStation = "cergy+le+haut";
    const url = `/next-trains/rers/A/${fromStation}/${toStation}`;
    const response = await api.get(url).expect(200);
    expect(response.body.departures).toMatchSnapshot();
    expect(response.body.departures.length).toBe(2);
  });
});
