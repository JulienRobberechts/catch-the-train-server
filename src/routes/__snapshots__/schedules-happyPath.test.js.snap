// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /next-train/{network}/{line}/{station}?missions=... should return 'UPAC' and 'ZEBU' missions for chatelet 1`] = `
Array [
  Object {
    "departureTime": "2020-03-10T09:24:00+01:00",
    "displayAttributes": "09:24",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0924",
  },
  Object {
    "departureTime": "2020-03-10T09:27:00+01:00",
    "displayAttributes": "09:27",
    "displayDestination": "Saint-Germain-en-Laye",
    "mission": "ZEBU",
    "trainCode": "0927",
  },
  Object {
    "departureTime": "2020-03-10T09:29:00+01:00",
    "displayAttributes": "09:29",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0929",
  },
  Object {
    "departureTime": "2020-03-10T09:35:00+01:00",
    "displayAttributes": "09:35",
    "displayDestination": "Saint-Germain-en-Laye",
    "mission": "ZEBU",
    "trainCode": "0935",
  },
  Object {
    "departureTime": "2020-03-10T09:47:00+01:00",
    "displayAttributes": "09:47",
    "displayDestination": "Saint-Germain-en-Laye",
    "mission": "ZEBU",
    "trainCode": "0947",
  },
]
`;

exports[`GET /next-train/{network}/{line}/{station}?missions=... should return 'UPAC' missions for chatelet 1`] = `
Array [
  Object {
    "departureTime": "2020-03-10T09:24:00+01:00",
    "displayAttributes": "09:24",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0924",
  },
  Object {
    "departureTime": "2020-03-10T09:29:00+01:00",
    "displayAttributes": "09:29",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0929",
  },
]
`;

exports[`GET /next-train/{network}/{line}/{station}?missions=... should return all missions for chatelet 1`] = `
Array [
  Object {
    "departureTime": "2020-03-10T09:24:00+01:00",
    "displayAttributes": "09:24",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0924",
  },
  Object {
    "departureTime": "2020-03-10T09:27:00+01:00",
    "displayAttributes": "09:27",
    "displayDestination": "Saint-Germain-en-Laye",
    "mission": "ZEBU",
    "trainCode": "0927",
  },
  Object {
    "departureTime": "2020-03-10T09:29:00+01:00",
    "displayAttributes": "09:29",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0929",
  },
  Object {
    "departureTime": "2020-03-10T09:35:00+01:00",
    "displayAttributes": "09:35",
    "displayDestination": "Saint-Germain-en-Laye",
    "mission": "ZEBU",
    "trainCode": "0935",
  },
  Object {
    "departureTime": "2020-03-10T09:37:00+01:00",
    "displayAttributes": "09:37",
    "displayDestination": "Poissy",
    "mission": "TEDI",
    "trainCode": "0937",
  },
  Object {
    "departureTime": "2020-03-10T09:47:00+01:00",
    "displayAttributes": "09:47",
    "displayDestination": "Saint-Germain-en-Laye",
    "mission": "ZEBU",
    "trainCode": "0947",
  },
  Object {
    "departureTime": "2020-03-10T19:12:00+01:00",
    "displayAttributes": "19:12",
    "displayDestination": "Poissy",
    "mission": "TEDI",
    "trainCode": "1912",
  },
  Object {
    "departureTime": "2020-03-11T00:15:00+01:00",
    "displayAttributes": "00:15",
    "displayDestination": "Poissy",
    "mission": "TEDI",
    "trainCode": "2415",
  },
]
`;

exports[`GET schedule by destination should return 2 departures from chatelet+les+halles to cergy+le+haut 1`] = `
Array [
  Object {
    "departureTime": "2020-03-10T09:24:00+01:00",
    "displayAttributes": "09:24",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0924",
  },
  Object {
    "departureTime": "2020-03-10T09:29:00+01:00",
    "displayAttributes": "09:29",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0929",
  },
]
`;

exports[`GET schedule by destination should return 5 departures from chatelet+les+halles to maisons+laffitte 1`] = `
Array [
  Object {
    "departureTime": "2020-03-10T09:24:00+01:00",
    "displayAttributes": "09:24",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0924",
  },
  Object {
    "departureTime": "2020-03-10T09:29:00+01:00",
    "displayAttributes": "09:29",
    "displayDestination": "Cergy-Le-Haut",
    "mission": "UPAC",
    "trainCode": "0929",
  },
  Object {
    "departureTime": "2020-03-10T09:37:00+01:00",
    "displayAttributes": "09:37",
    "displayDestination": "Poissy",
    "mission": "TEDI",
    "trainCode": "0937",
  },
  Object {
    "departureTime": "2020-03-10T19:12:00+01:00",
    "displayAttributes": "19:12",
    "displayDestination": "Poissy",
    "mission": "TEDI",
    "trainCode": "1912",
  },
  Object {
    "departureTime": "2020-03-11T00:15:00+01:00",
    "displayAttributes": "00:15",
    "displayDestination": "Poissy",
    "mission": "TEDI",
    "trainCode": "2415",
  },
]
`;
